lua/neogit/popups/branch/actions.lua

```diff
@@ -22,6 +22,40 @@ local function parse_remote_branch_name(ref)
   return remote, branch_name
 end
 
+M.spin_off_branch = operation("spin_off_branch", function()
+  if #git.repo.staged.items > 0 or #git.repo.unstaged.items > 0 then
```

Comment:
You can extract 
```lua
#git.repo.staged.items > 0 or #git.repo.unstaged.items > 0
```
to `lua/neogit/lib/git/status.lua` as `M.is_dirty()` (or something like that)

Reply:
Good call 👍🏼